File: marcs-cakewalk/marcs-cakewalk.js
"use strict";

const fs = require("fs");

process.stdin.resume();
process.stdin.setEncoding("utf-8");

let inputString = "";
let currentLine = 0;

process.stdin.on("data", function (inputStdin) {
  inputString += inputStdin;
});

process.stdin.on("end", function () {
  inputString = inputString.split("\n");

  main();
});

function readLine() {
  return inputString[currentLine++];
}

/*
 * Complete the 'marcsCakewalk' function below.
 *
 * The function is expected to return a LONG_INTEGER.
 * The function accepts INTEGER_ARRAY calorie as parameter.
 */

function marcsCakewalk(calorie) {
  let miles = 0;
  calorie.sort((a, b) => a - b);
  let j = 0;
  for (let i = calorie.length - 1; i >= 0; i--) {
    miles += Math.pow(2, j++) * calorie[i];
  }
  return miles;
}

function main() {
  const ws = fs.createWriteStream(process.env.OUTPUT_PATH);

  const n = parseInt(readLine().trim(), 10);

  const calorie = readLine()
    .replace(/\s+$/g, "")
    .split(" ")
    .map((calorieTemp) => parseInt(calorieTemp, 10));

  const result = marcsCakewalk(calorie);

  ws.write(result + "\n");

  ws.end();
}

File: marcs-cakewalk/MarcsCakewalk.java
import java.io.*;
import java.util.*;
import java.util.stream.*;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'marcsCakewalk' function below.
     *
     * The function is expected to return a LONG_INTEGER.
     * The function accepts INTEGER_ARRAY calorie as parameter.
     */

    public static long marcsCakewalk(List<Integer> calorie) {
        long miles = 0;
        Collections.sort(calorie);
        int j = 0;
        for (int i = calorie.size() - 1; i >= 0; i--) {
            miles += (Math.pow(2, j++) * calorie.get(i));
        }
        return miles;
    }

}

public class MarcsCakewalk {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<Integer> calorie = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                .map(Integer::parseInt)
                .collect(toList());

        long result = Result.marcsCakewalk(calorie);

        bufferedWriter.write(String.valueOf(result));
        bufferedWriter.newLine();

        bufferedReader.close();
        bufferedWriter.close();
    }
}
